pkgs/by-name/ut/ut1999/package.nix

```diff
@@ -0,0 +1,122 @@
+{ lib, stdenv, requireFile, autoPatchelfHook, fetchurl,
+libgcc, wxGTK32, innoextract, makeWrapper,
+libGL, SDL2, openal, libmpg123, libxmp }:
+
+let
+  oldUnrealPatch = stdenv.mkDerivation {
+    name = "ut1999-oldunreal-patch";
+    version = "469d";
+    sourceRoot = ".";
+    src = fetchurl {
+      url = "https://github.com/OldUnreal/UnrealTournamentPatches/releases/download/v469d/OldUnreal-UTPatch469d-Linux-amd64.tar.bz2";
+      hash = "sha256-aoGzWuakwN/OL4+xUq8WEpd2c1rrNN/DkffI2vDVGjs=";
+    };
+
+    buildInputs = [
+      libgcc
+      stdenv.cc.cc
+      wxGTK32
+    ];
+
+    nativeBuildInputs = [
+      autoPatchelfHook
+      makeWrapper
+    ];
+
+    installPhase = ''
+      runHook preInstall
+
+      mkdir -p $out
+      ls -alh
+      cp -r ./* $out
+
+      runHook postInstall
+    '';
+
+    dontBuild = true;
+  };
+
+  unpackGog = stdenv.mkDerivation {
+    name = "ut1999-gog";
+    version = "2.0.0.5";
+
+    src = requireFile rec {
+      name = "setup_ut_goty_2.0.0.5.exe";  # FIXME: use official name
+      sha256 = "00v8jbqhgb1fry7jvr0i3mb5jscc19niigzjc989qrcp9pamghjc";
+      message = ''
+        Unreal Tournament 1999 requires the official GOG package, version 2.0.0.5.
+
+        Once you download the file, run the following command:
+
+        nix-prefetch-url file://\$PWD/${name}
+      '';
+    };
+
+    buildInputs = [
+      innoextract
+    ];
+
+    unpackPhase = ''
+      innoextract --extract --exclude-temp "$src"
```

Comment:
Run the hooks.

Reply:
@Atemu , I don't seem to find the corresponding hooks for `unpackPhase`. Maybe not needed?

Reply:
The stdenv attempts to run the hooks if they exist.